leptos_dom/src/callback.rs

```diff
@@ -0,0 +1,245 @@
+// inspired by:
+// https://github.com/lpotthast/leptonic/blob/f8a270ae5512561a92f5deca057f7999940de11b/leptonic/src/callback.rs
+
+use crate::{AnyElement, ElementDescriptor, HtmlElement, IntoView, View};
+use leptos_reactive::StoredValue;
+use std::{rc::Rc, sync::Arc};
+
+/// A wrapper trait for calling callbacks.
+pub trait Callable<In, Out = ()> {
+    /// calls the callback with the specified argument.
+    fn call(&self, input: In) -> Out;
+}
+
+/// The most basic leptos callback type.
+/// It is intended to make passing a function to your component easy.
+///
+/// # Usage when you write a component
+/// ```
+/// #[component]
+/// fn MyComponent(#[prop(into)] render_number: Callback<i32, String>) {
+///     view! {
+///         <div>
+///             {render_number.call(42)}
+///         </div>
+///     }
+/// }
+/// ```
+/// # Usage when you use your component
+/// ```
+/// view! {
+///     <MyComponent callback=move |x| x.to_string()/>
+/// }
+/// ```
+///
+/// Note that in this scenario, you should use a generic component instead.
+/// You should use callback mainly for optional generic props.
+///
+/// # Cloning
+/// The default [Callback] type can be cloned cheaply but cannot be copied.
+/// If you want to use the callback in a lot of closures, you will have to clone it each time.
+/// To avoid that, see [StoredCallback]
+#[derive(Clone)]
```

Comment:
This should be switched to a manual `Clone` implementation; deriving `Clone` means that `Callback` is only clone if `In` and `Out` are clone, when in fact a manual implementation lets you just clone the inner `Rc`.